### ISSUE TITLE
Highlight required fields

### ISSUE THREAD

**Issue by saulipurhonen** [CSCfi/metadata-submitter-frontend#291]
#### Describe the bug

Highlighting of required fields doesn't work properly.
It seems that `WizardJSONSchemaParser` doesn't handle `required` attribute as it should.

#### To Reproduce

1. Create new submission
2. Click Analysis
3.  `Analysis Type` dropdown isn't highlighted

#### Expected behavior

All required fields should be highlighted.

**Comment by saulipurhonen** [comment]
I'm having some problems with `Experiment` object type. 
We have `design` as required property, but:

There isn't any required field under `design`. Same situation with referenced fields. What's the purpose of having `design` property as required field?

Another one one that I'm not able to get is if `Experiment`/`Study Reference` is missing and `Accession Id` field we get a response from backend that `Validation failed, details: Provided input does not seem correct because: ''design' is a required property'`. I think it would make sense to have a message that points to `Study Reference` instead.

I've been trying to study the model from [the metadata-submitter docs](https://metadata-submitter.readthedocs.io/en/docs-metadata-and-auth/metadata.html#relationships-between-objects), but haven't found any clues yet.

@blankdots, any ideas how to go from here?

**Comment by saulipurhonen** [comment]
For the original issue the problem is that `oneOf` fields don't have handling for `required` attribute.

**Comment by blankdots** [comment]
@saulipurhonen sometimes whole sections of a from are required and not individual fields. Just in case I will review the schemas again and get the required fields, or remove some of them.

**Comment by blankdots** [comment]
the schema of `Experiment` is such that `design` and `study reference` are required, but is fields are not required so in practice an empty object could be sent, although not ideal. For `design` I identified some required fields and I placed them in `develop` backend branch and should show in UI (hopefully).

the way I see it there are 2 options when we have sections that are required but their fields are not:
1. we send an empty object - not ideal, but a good start
2. we restrict it the UI that at least one field is filled in - so that we don't break schemas

If we were to add more restrictions to the schemas it will break the XML Schema <-> JSON Schema parity 

the fields which I identified are:
1. `experiment` -> `study reference`
2. `policy` -> `DAC reference`
3. `run` -> `experiment reference`
